assignment/assignment.hs

```diff
@@ -14,9 +14,14 @@ import Data.Char(intToDigit)
 -- The build in function `reverse` takes a list and returns
 -- a new list with the elements in reverse order.
 --
--- Implement a version vcalled `myreverse` using `foldl`.
+-- Implement a version called `myreverse` using `foldl`.
+
+swap :: [a] -> a -> [a] 
+swap h r = [r] ++ h
+myreverse :: [a] -> [a]
+myreverse [] = []
+myreverse (l:ls) = foldl swap [l] ls
 
```

Comment:
When you use a fold, you expect it to take care of the empty list case for you, so writing the [] = [] case might have set some alarm bells ringing.

The function passed to fold received the result so far and the next item. If you simply add that item to the front of the list so far, then subsequent items will get put in front of it, reversing the list. So you could write

~~~ hs
myReverse =  foldl (\lst h -> h:=lst) []
~~

or even

~~~ hs
myReverse = foldl (flip (:)) []
~~~